lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -121,10 +131,7 @@ def getvm(self, name=None, uuid=None, folder=None):
         if uuid:
             vm = find_vm_by_id(self.content, vm_id=uuid, vm_id_type="uuid")
         elif folder:
-            # Build the absolute folder path to pass into the search method
-            if not self.params['folder'].startswith('/'):
-                self.module.fail_json(msg="Folder %(folder)s needs to be an absolute path, starting with '/'." % self.params)
-            searchpath = '%(datacenter)s%(folder)s' % self.params
+            searchpath = self.params['folder']
```

Comment:
I don't think there will be any backwards compatibility issues with this change. User would not see any behavior change as we are internally changing searchpath. I have added examples for folder path in documentation section which will give a clear idea about this change. 
@jctanner Would you like add something to this ?